activemq-http/src/main/java/org/apache/activemq/transport/ws/WSTransportServer.java

```diff
@@ -96,7 +96,13 @@ protected void doStart() throws Exception {
 
         contextHandler.setAttribute("acceptListener", getAcceptListener());
 
-        server.start();
+        ClassLoader original = Thread.currentThread().getContextClassLoader();
+        Thread.currentThread().setContextClassLoader(WSTransportServer.class.getClassLoader());
```

Comment:
This could be problematic in OSGi/Karaf generally speaking. In this specific it should be ok as the `WSTransportServer` is in the `activemq-osgi` bundle.